Dockerfile

```diff
@@ -1,18 +1,57 @@
-FROM ubuntu:xenial
+# syntax=docker/dockerfile:experimental
 
-RUN apt-get update && apt-get install --yes nginx
+# Build stage: Install ruby dependencies
+# ===
+FROM ubuntu:bionic AS ruby-dependencies
+RUN apt-get update && apt-get install --no-install-recommends -y build-essential ruby-dev ruby-bundler
+
+# Build stage: Install yarn dependencies
+# ===
+FROM node:10-slim AS yarn-dependencies
+WORKDIR /srv
+ADD package.json .
+RUN --mount=type=cache,target=/usr/local/share/.cache/yarn yarn install
 
-# Set git commit ID
-ARG REVISION_ID
-RUN test -n "${REVISION_ID}"
 
-# Copy over files
+# Build stage: Run "yarn run build-jekyll"
+# ===
+FROM ruby-dependencies AS build-jekyll
 WORKDIR /srv
-ADD _site .
-ADD nginx.conf /etc/nginx/sites-enabled/default
-RUN sed -i "s/~REVISION_ID~/${REVISION_ID}/" /etc/nginx/sites-enabled/default
+ADD . .
+RUN bundle install
+RUN bundle exec jekyll build
 
-STOPSIGNAL SIGTERM
+# Build stage: Run "yarn run build-js"
+# ===
+FROM yarn-dependencies AS build-js
+ADD . .
+RUN yarn run build-js
 
-CMD ["nginx", "-g", "daemon off;"]
 
+# Build stage: Run "yarn run build-css"
+# ===
+FROM yarn-dependencies AS build-css
+ADD . .
+RUN yarn run build-css
+
+# Build the production image
+# ===
+FROM ubuntu:bionic
+RUN apt-get update && apt-get --no-install-recommends --yes install nginx
+
+# Set up environment
+ENV LANG C.UTF-8
+WORKDIR /srv
+
+# Import code, build assets and mirror list
+RUN rm -rf package.json yarn.lock .babelrc webpack.config.js
+COPY --from=build-jekyll srv/_site .
+COPY --from=build-css srv/css css
+COPY --from=build-js srv/js js
+
+ARG BUILD_ID
+ADD nginx.conf /etc/nginx/sites-enabled/default
+RUN sed -i "s/~BUILD_ID~/${BUILD_ID}/" /etc/nginx/sites-enabled/default
+
```

Comment:
Add this line: `STOPSIGNAL SIGTERM`
https://ubuntu.com/blog/avoiding-dropped-connections-in-nginx-containers-with-stopsignal-sigquit

ping @nottrobin  :)